### PR TITLE
Add optional custom 503 page in Application struct

### PR DESCRIPTION
**Pull request by NotBad4U**
work for #508

### CODE REVIEW

**Comment by NotBad4U** [comment]
Ready to a review

**Comment by NotBad4U** [comment]
Do we allow people to set the custom 503 page with `sozuctl` or can I just [sozuctl add_application()](https://github.com/sozu-proxy/sozu/blob/4eacbf6fc619570620323137c290a1477aad2d93/ctl/src/command.rs#L738) set `answer_503` to `None` ?

File: lib/src/network/http.rs [hunk 1]
```diff
@@ -884,14 +899,30 @@ impl ServerConfiguration {
...
         client.set_answer(DefaultAnswerStatus::Answer503, answer);
         return Err(ConnectionError::NoBackendAvailable);
       }
     }
 
     Ok(())
   }
+
+  fn get_service_unavailable_answer(&self, app_id: &str, listen_token: &Token) -> Rc<Vec<u8>> {
+    if let Some(ref custom_answers) = self.custom_answers.get(app_id) {
```
Comment (Geal) [inline lib/src/network/http.rs]: maybe replace the two if let with a `and_then`

**Comment by Geal** [comment]
please implement it for rustls too

**Comment by Geal** [comment]
the custom answers should be remove in `remove_application`

**Comment by NotBad4U** [comment]
> please implement it for rustls too

Done in db5c416

**Comment by NotBad4U** [comment]
> the custom answers should be remove in `remove_application`

Done in `8b61af6`

**Pull request merged**